### PR TITLE
Add initial threat model

### PR DESCRIPTION
**Pull request by tomwillis608**
A little help from an AI

### CODE REVIEW

File: docs/threat-model.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+### Threat
+
+Improper handling of user credentials, such as GitHub username and password, leading to potential exposure or compromise.
+
+### Mitigation
+
+Implement secure credential management practices, such as securely storing credentials, avoiding hard-coding them within
+the source code, and utilizing secure credential storage mechanisms like environment variables or secret management
+solutions. Ensure that sensitive credentials, such as the GitHub username and password, are properly protected and encrypted.
```
Comment (halprin) [inline docs/threat-model.md:46]: You have two threats and two mitigations in this section that are awfully similar.

Comment (tomwillis608) [reply docs/threat-model.md]: I made some changes based on this feedback, Peter.

File: docs/threat-model.md [hunk 2]
```diff
@@ -132,17 +142,11 @@ Authorized users with malicious intent exploiting their privileges to misuse or
...
 
-Insufficient input validation for the `connect_to_github` function, allowing for potentially malicious inputs.
+Users of the library are responsible within their own sphere of influence.
 
-### Mitigation
-
-Implement robust input validation and strict parsing of inputs for the `connect_to_github` function. Validate and sanitize
-user-provided inputs to mitigate potential security risks, such as ensuring the username and password inputs adhere to
-the expected format and length.
+## Lack of Input Validation for Connect to GitHub Function
```
Comment (halprin) [inline docs/threat-model.md:149]: This turned into an empty section here.  Is that on purpose?

**Review by halprin — COMMENTED** [review]
Thanks.  One more thing seemed to come out of your changes.

File: docs/threat-model.md (hunk 2 above)
Comment (tomwillis608) [reply docs/threat-model.md:149]: This is a missed delete.  This risk was redundant with an earlier risk.

Comment (tomwillis608) [reply docs/threat-model.md]: @halprin please re-review.  this is a really strict repo, apparently

Comment (halprin) [reply docs/threat-model.md]: You could just tell me to stuff it.  😉

**Review by halprin — APPROVED** [review]
Beautiful!

**Pull request merged**